### ISSUE TITLE
tf.norm causes nan in the gradients

### ISSUE THREAD

**Issue by Andrea94c** [stallmo/dcn-deep-clustering-networks#1]
I found that tf.norm can cause some gradients to be nan. 

I replaced tf.norm with a safer norm function, it works fine for me now: 

def safe_norm(x, epsilon=1e-12, axis=None):
    return tf.sqrt(tf.reduce_sum(x ** 2, axis=axis) + epsilon)

Note: I'm not super familiar with this tensorflow version: nans can be caused elsewhere.
